src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts

```diff
@@ -118,26 +118,30 @@ export class StickyScrollWidget extends Disposable implements IOverlayWidget {
 		return this._lineNumbers.length;
 	}
 
-	getStickyLineForLine(lineNumber: number): RenderedStickyLine | undefined {
-		return this._stickyLines.find(stickyLine => stickyLine.lineNumber === lineNumber);
+	getRenderedStickyLine(lineNumber: number): RenderedStickyLine | undefined {
+		return this._renderedStickyLines.find(stickyLine => stickyLine.lineNumber === lineNumber);
 	}
 
 	getCurrentLines(): readonly number[] {
 		return this._lineNumbers;
 	}
 
-	setState(state: StickyScrollWidgetState | undefined, foldingModel: FoldingModel | null, rebuildFromLine: number = 0): void {
-		if ((!this._previousState && !state) || (this._previousState && this._previousState.equals(state))) {
+	setState(_state: StickyScrollWidgetState | undefined, foldingModel: FoldingModel | null, _rebuildFromLine?: number): void {
```

Comment:
I don't understand why you can still pass in `rebuildFromLine` when you can compute it here anyway.

Reply:
That parameter is a bit different because it forces the rebuild from line. So for example when the tokens change, we want to force rebuild the whole sticky widget because all of the lines are stale, even if we share part of the state with the previous state.

Perhaps in order to make it less confusing, I can just add a boolean of type `forceFullRebuild` which will just force set the subsequent rebuildFromLine parameter to 0. I'll do this in a separate PR.